src/services/directoryServices/MediaDirectoryService.js

```diff
@@ -40,39 +38,26 @@ class MediaDirectoryService {
     const { private: isPrivate } = await this.gitHubService.getRepoInfo(
       sessionData
     )
-    const files = await this.listWithDefault(sessionData, { directoryName })
-
-    const resp = []
-    for (const curr of files) {
-      if (curr.type === "dir") {
-        resp.push({
-          name: curr.name,
-          type: "dir",
-        })
-      }
-      if (curr.type !== "file" || curr.name === PLACEHOLDER_FILE_NAME) continue
-      const fileData = {
-        mediaUrl: `https://raw.githubusercontent.com/${GITHUB_ORG_NAME}/${siteName}/staging/${curr.path
-          .split("/")
-          .map((v) => encodeURIComponent(v))
-          .join("/")}${curr.path.endsWith(".svg") ? "?sanitize=true" : ""}`,
-        name: curr.name,
-        sha: curr.sha,
-        mediaPath: `${directoryName}/${curr.name}`,
-        type: curr.type,
-      }
-      if (mediaType === "images" && isPrivate) {
-        // Generate blob url
-        const imageExt = curr.name.slice(curr.name.lastIndexOf(".") + 1)
-        const contentType = `image/${imageExt === "svg" ? "svg+xml" : imageExt}`
-        const { content } = await this.gitHubService.readMedia(sessionData, {
-          fileSha: curr.sha,
+    const files = (
+      await this.listWithDefault(sessionData, { directoryName })
+    ).filter(
+      (file) =>
+        (file.type === "file" || file.type === "dir") &&
+        file.name !== PLACEHOLDER_FILE_NAME
+    )
+
+    const resp = await Promise.all(
```

Comment:
That makes a lot of sense, but I think this might be better to handle within `getMediaFileInfo` instead? I've inserted a try/catch block in https://github.com/isomerpages/isomercms-backend/pull/740/commits/bb93aa2e83e8c35c193924db891c772eb5308da3 instead, as we should still need some file details which are extracted by the method, and we just need to convert the `mediaUrl` to a generic value to be able to handle individual file failures

As discussed offline, the frontend currently has a generic image when it is unable to retrieve an image - the user can still attempt to fetch the actual file data again by clicking on the image, which loads up a modal where the image details are refetched.

Reply:
Cool sounds good

Reply:
Are we still keeping Promise.all though? So we want the request to fail if any of the loading fails?